packages/algob/src/internal/deployer.ts

```diff
@@ -114,11 +117,30 @@ class DeployerBasicMode {
    * @returns {LogicSig} multi signed logic signature from assets/<file_name>.msig
    */
   async loadMultiSig (name: string, scParams: Object): Promise<LogicSig> {
-    const [tealFile, Msig] = await readMsigFromFile(name) as string; // Get tealFile name, Msig object from .mlsig
-    const lsig = await this.loadLsig(tealFile, scParams); // Load lsig from .teal (getting logic part from lsig)
-    const decodedMsig = await decodeMsigObj(Msig);
-    lsig.msig = {};
-    Object.assign(lsig.msig, decodedMsig);
+    const lsig = await getLsig(name, scParams, this.algoOp.algodClient); // get lsig from .teal (getting logic part from lsig)
+    const Msig = await readMsigFromFile(name); // Get decoded Msig object from .msig
+    Object.assign(lsig.msig = {}, Msig);
+    return lsig;
+  }
+
+  /**
+   * Description : loads multisigned logic signature from .msig file
+   * @param {string} name filename
+   * @returns {LogicSig} multi signed logic signature from assets/<file_name>.msig
+   */
+  async loadBinaryMultiSig (name: string): Promise<LogicSig> {
+    // get logic signature from file and decode it
+    const data = await readBinaryMultiSig(name);
+    const program = new Uint8Array(Buffer.from(data, 'base64'));
+    const logicSignature = decode(program) as LogicSig;
+
+    // dummy logic signature
+    const dummyProgram = new Uint8Array(56);
+    dummyProgram.fill(0);
+    const lsig = new logicsig.LogicSig(dummyProgram, []);
+
+    lsig.logic = logicSignature.l; // assign logic
+    Object.assign(lsig.msig = {}, logicSignature.msig); // assign signature
     return lsig;
```

Comment:
what are we assigning only `msig` and `logic` part of raw multisig ?
is this desirable for user ?

Reply:
we should use `Object.assign(lsig, logicSignature);`

Reply:
I agree. But the keys are different for decoded `lsig` from file and the dummy one (check line 142). Continuing on discord.

Reply:
yes, we have to fix that else this is incomplete solution

Reply:
Working on it.

Reply:
Is this solved now?

Reply:
Yes